com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -314,40 +314,52 @@ internal bool HasPrefab(NetworkObject.SceneObject sceneObject)
         }
 
         /// <summary>
-        /// Should only run on the client
+        /// Creates a local NetowrkObject to be spawned.
         /// </summary>
-        internal NetworkObject CreateLocalNetworkObject(bool isSceneObject, uint globalObjectIdHash, ulong ownerClientId, ulong? parentNetworkId, int? networkSceneHandle, Vector3? position, Quaternion? rotation, bool isReparented = false)
+        /// <remarks>
+        /// For most cases this is client-side only, with the exception of when the server
+        /// is spawning a player.
+        /// </remarks>
+        internal NetworkObject CreateLocalNetworkObject(NetworkObject.SceneObject sceneObject)
         {
             NetworkObject parentNetworkObject = null;
+            var globalObjectIdHash = sceneObject.Header.Hash;
+            var position = sceneObject.Header.HasTransform ? sceneObject.Transform.Position : default;
+            var rotation = sceneObject.Header.HasTransform ? sceneObject.Transform.Rotation : default;
+            var isReparented = sceneObject.Header.IsReparented;
+            var parentNetworkId = sceneObject.Header.HasParent ? sceneObject.ParentObjectId : default;
+            var worldPositionStays = sceneObject.Header.HasParent ? sceneObject.WorldPositionStays : true;
 
-            if (parentNetworkId != null && !isReparented)
+            if (sceneObject.Header.HasParent)
             {
-                if (SpawnedObjects.TryGetValue(parentNetworkId.Value, out NetworkObject networkObject))
+                if (SpawnedObjects.TryGetValue(sceneObject.ParentObjectId, out NetworkObject networkObject))
                 {
                     parentNetworkObject = networkObject;
                 }
-                else
+                else if (isReparented) // TODO: Not sure we need this warning any longer
                 {
-                    if (NetworkLog.CurrentLogLevel <= LogLevel.Normal)
+                    if (NetworkLog.CurrentLogLevel <= LogLevel.Developer)
                     {
                         NetworkLog.LogWarning("Cannot find parent. Parent objects always have to be spawned and replicated BEFORE the child");
                     }
                 }
             }
 
-            if (!NetworkManager.NetworkConfig.EnableSceneManagement || !isSceneObject)
+            if (!NetworkManager.NetworkConfig.EnableSceneManagement || !sceneObject.Header.IsSceneObject)
```

Comment:
Yeah the name actually I think we could change...this seems to be what it was called for sometime now.
I seem to remember creating like a NetworkObject.Serialize and NetworkObject.Deserailize a long time ago and this turned into "SceneObject" from the message itself.

Reply:
@JesseOlmer 
This was the only area that I felt could be another PR since this one has started to become relatively large.
Would you like me to come up with a different name for NetworkObject.SceneObject in this PR? 
If so, do you have any suggestions for the name change?